src/datasets/load.py

```diff
@@ -133,6 +135,71 @@ def import_main_class(module_path, dataset=True) -> Optional[Union[Type[DatasetB
     return module_main_cls
 
 
+class _InitializeConfiguredDatasetBuilder:
+    """
```

Comment:
heh unfortunately, we need `BUILDER_CONFIGS` to be a class attribute, because it's used in `DatasetBuilder.__init__`, in `_create_builder_config()`, so it should be filled properly before initialization.

Reply:
> heh unfortunately, we need BUILDER_CONFIGS to be a class attribute, because it's used in DatasetBuilder.__init__(), in _create_builder_config(), so it should be filled properly before initialization.

I think then we can do the following:

```python
configured_builder = DatasetBuilder.__new__(cls)
configured_builder.BUILDER_CONFIGS = builder_configs
configured_builder.__init__(args, kwargs)
```

Reply:
hm then we have [`DatasetBuilder.builder_configs`](https://github.com/huggingface/datasets/blob/main/src/datasets/builder.py#L539) class property which is created from builder's *class* `BUILDER_CONFIGS` and it would be empty dict with this approach which then leads to indexing error while config creation. I'm not sure it can be easily adjusted in a straightforward way either, do you have an idea?

Reply:
I don't think there is a backward-compatible solution to this, so it's probably best to stick to the current solution.